components/camel-cmis/src/main/java/org/apache/camel/component/cmis/CMISProducer.java

```diff
@@ -57,12 +62,12 @@ public CMISEndpoint getEndpoint() {
     }
 
     public void process(Exchange exchange) throws Exception {
-        CmisObject cmisObject = createNode(exchange);
-        log.debug("Created node with id: {}", cmisObject.getId());
 
-        // copy the header of in message to the out message
-        exchange.getOut().copyFrom(exchange.getIn());
-        exchange.getOut().setBody(cmisObject.getId());
+        CamelCMISActions action = exchange.getIn().getHeader(CamelCMISConstants.CMIS_ACTION, CamelCMISActions.class);
+
+        Class[] paramMethod = null;
```

Comment:
This is working, but the correct one looks like Exchange.class...

Reply:
paramMethod should be Exchange.class type ?

Reply:
Yes, Class[] paramMethod = {Exchange.class};